internal/provider/svm/svm_qos_policy_activation_resource.go

```diff
@@ -0,0 +1,407 @@
+func NewSvmQosPolicyActivationResource() resource.Resource {
+	return &SvmQosPolicyActivationResource{
+		config: connection.ResourceOrDataSourceConfig{
+			Name: "svm_qos_policy_activation",
+		},
+	}
+}
+
+// SvmQosPolicyActivationResource defines the resource implementation.
+type SvmQosPolicyActivationResource struct {
+	config connection.ResourceOrDataSourceConfig
+}
+
+// SvmQosPolicyActivationResourceModel describes the resource data model.
+type SvmQosPolicyActivationResourceModel struct {
+	CxProfileName types.String                                  `tfsdk:"cx_profile_name"`
+	Svm           *SvmQosPolicyActivationSvmResourceModel       `tfsdk:"svm"`
+	QoSPolicy     *SvmQosPolicyActivationQosPolicyResourceModel `tfsdk:"qos_policy"`
+}
+
+type SvmQosPolicyActivationSvmResourceModel struct {
+	ID   types.String `tfsdk:"id"`
+	Name types.String `tfsdk:"name"`
+}
+
+type SvmQosPolicyActivationQosPolicyResourceModel struct {
+	ID   types.String `tfsdk:"id"`
+	Name types.String `tfsdk:"name"`
+}
+
+// Metadata returns the resource type name.
+func (r *SvmQosPolicyActivationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + r.config.Name
+}
+
+// Schema defines the schema for the resource.
+func (r *SvmQosPolicyActivationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Resource to activate QoS Policy for SVM",
+
+		Attributes: map[string]schema.Attribute{
+			"cx_profile_name": schema.StringAttribute{
+				MarkdownDescription: "Connection profile name",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"svm": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"name": schema.StringAttribute{
+						MarkdownDescription: "The name of the SVM",
+						Optional:            true,
+						Computed:            true,
+					},
+					"id": schema.StringAttribute{
+						MarkdownDescription: "The unique identifier of the SVM",
+						Optional:            true,
+						Computed:            true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+				},
+				MarkdownDescription: "SVM to manage (name or UUID)",
+				Required:            true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.RequiresReplace(),
+				},
+			},
+			"qos_policy": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"name": schema.StringAttribute{
+						MarkdownDescription: "The QoS policy group name",
+						Optional:            true,
+						Computed:            true,
+					},
+					"id": schema.StringAttribute{
```

Comment:
We try to use name all the time, as UUID is not something user friendly, and name itself can handle all functionality, so we have used name in all our Ansible/Terrafrom modules/resources. Unless you have an explicit case, let us know